client/components/Editor/plugins/discussions/util.ts

```diff
@@ -32,15 +32,23 @@ export const mapDiscussionSelectionThroughSteps = (
 	if (isEmptySelection(selection)) {
 		return null;
 	}
-	const mapping = new Mapping(steps.map((step) => step.getMap()));
 	const { from, to } = getRangeFromSelection(selection!);
 	// This matches the default mapping of inline Decorations, which we wish to emulate here.
 	// The default Selection mapping behavior is too greedy; for instance, if a selection is made
 	// at the beginning of the document, any text inserted before that selection is also mapped
 	// into that selection, which is not what we want for anchored Discussions.
 	// See: https://github.com/ProseMirror/prosemirror-view/blob/49889c576bc450a3008117249f9ef9beb6bf2969/src/decoration.js#L36
-	const nextFrom = mapping.map(from, 1);
-	const nextTo = mapping.map(to, -1);
+	const { from: nextFrom, to: nextTo } = steps.reduce(
```

Comment:
Sadly we cannot use this approach from the previous revision:
```
const mapping = new Mapping(steps.map((step) => step.getMap()));
// map positions with mapping.map()
```
Instead, we need to map the positions through each Step, one at a time, or we'll lose important information. For instance, the above `Mapping` treats the act of erasing content and then adding new content as a simple replacement, which improperly preserves the highlight on the old content where it may be entirely irrelevant. I _think_ that this is the behavior we want.

(This is only important for Release-side discussions, which get mapped through many steps at once — draft-side discussions go through the same function, but typically `steps.length === 1` in that case and the result is the same)

Reply:
Agreed that that is the behavior we want.